docs/the-new-architecture/pure-cxx-modules.md

```diff
@@ -0,0 +1,472 @@
+- Specify where CMake can find the additional header files. Also in this case we need to crawl back from the `jni` folder.
+
+#### 2. Modify gradle.build to link the C++ code
+
+Gradle is the tool that orchestrates the Android build. We need to tell it where it can find the `CMake` files to build the Turbo Native Module.
+
+1. Open the `SampleApp/android/app/build.gradle` file
+2. Add the following block to the gradle file:
+
+```diff
+    if (hermesEnabled.toBoolean()) {
+        implementation("com.facebook.react:hermes-android")
+    } else {
+        implementation jscFlavor
+    }
+}
+
++ android {
++   externalNativeBuild {
++       cmake {
++           path "src/main/jni/CMakeLists.txt"
++       }
++   }
++}
+```
+
+This block tells the Gradle file where to look for the external native build files. The path is relative to the folder where the gradle.build file lives, so we need to add the path to the `CMakeLists.txt` files in the `jni` folder.
+
+#### 3. Register the new Turbo Native Module
+
+The final step is to register the new C++ Turbo Native Module in the runtime, so that when JS requires the C++ Turbo Native Module, the app knows where to find it and can return it.
+
+1. From the folder `SampleApp/android/app/src/main/jni`, run the following command:
+
+```sh
+curl -O https://raw.githubusercontent.com/facebook/react-native/v0.76.0/packages/react-native/ReactAndroid/cmake-utils/default-app-setup/OnLoad.cpp
+```
+
+2. Then, modify this file as it follows:
+
+```diff
+#include <DefaultComponentsRegistry.h>
+#include <DefaultTurboModuleManagerDelegate.h>
+#include <autolinking.h>
+#include <fbjni/fbjni.h>
+#include <react/renderer/componentregistry/ComponentDescriptorProviderRegistry.h>
+#include <rncore.h>
+
++ // Include the NativeSampleModule header
++ #include <NativeSampleModule.h>
+
+//...
+
+std::shared_ptr<TurboModule> cxxModuleProvider(
+    const std::string& name,
+    const std::shared_ptr<CallInvoker>& jsInvoker) {
+  // Here you can provide your CXX Turbo Modules coming from
+  // either your application or from external libraries. The approach to follow
+  // is similar to the following (for a module called `NativeCxxModuleExample`):
+  //
+  // if (name == NativeCxxModuleExample::kModuleName) {
+  //   return std::make_shared<NativeCxxModuleExample>(jsInvoker);
+  // }
+
++  // This code register the module so that when the JS side asks for it, the app can return it
++  if (name == NativeSampleModule::kModuleName) {
++    return std::make_shared<NativeSampleModule>(jsInvoker);
++  }
+
+  // And we fallback to the CXX module providers autolinked
+  return autolinking_cxxModuleProvider(name, jsInvoker);
+}
+
+// leave the rest of the file
+```
+
+These steps download the original OnLoad file from React Native, so that we can safely override it to load the C++ Turbo Native Module in the app.
```

Comment:
```suggestion
These steps download the original `OnLoad.cpp` file from React Native, so that we can safely override it to load the C++ Turbo Native Module in the app.
```